### ISSUE TITLE
Allow for overriding built-in rules

### ISSUE THREAD

**Issue by spadgos** [eslint/eslint#836]
A suggestion: if a custom rule exists in the `rulesdir` with the same name as one of the built-in rules, it should take precedence.

Use case: My project has a style of defining modules like so:

``` js
var MyModuleName,
    SomeRequire = require('some/module');

MyModuleName = module.exports = { ... }
```

When `no-unused-vars` has the option `{ "args": "all" }` set, it reports `MyModuleName` as an unused variable. Technically, yes it is, but for this project this is what is wanted.

I copied the `no-unused-vars` and modified it to allow the above style, but I had to give it another name, which is a bit messy, especially if you've got comment overrides in source files.

On one hand, it kind of is another rule so maybe it _should_ have a different name, but then again, it is applying the same basic rule but with an extra option, so it makes sense that a developer would expect `no-unused-vars` to be the correct name for it. It's now just how the `no-unused-vars` rule works for _this_ project.

Thoughts?
